.github/workflows/actions/build-upload-mithril-artifact/action.yml

```diff
@@ -19,11 +19,19 @@ runs:
       run: echo "short_sha=$(echo ${{ github.sha }} | cut -c1-7)" >> $GITHUB_OUTPUT
 
     - name: Add commit short sha to Cargo.tomls version
+      if: matrix.os != 'macos-14'
       shell: ${{ runner.os == 'Windows' && 'pwsh' || 'bash' }}
       run: |
         pip3 install toml
         python3 ./.github/workflows/scripts/edit-cargo-toml-version.py -l "${{ steps.label.outputs.short_sha }}"
 
+    - name: Add commit short sha to Cargo.tomls version on macOS-arm64 runners
+      if: matrix.os == 'macos-14'
+      shell: bash
+      run: |
+        pip3 install toml --break-system-packages
```

Comment:
Can we use this `--break-system-packages` option of `pip` for all the OS (and thus have a unique step for this operation)?

Reply:
Unfortunately, the build step for `Ubuntu x86` is failing because it's being executed on `Ubuntu 20.04.6`, which uses `Pip3 20.0.2`. The option --break-system-packages is only available from `Pip3 23.0.1`.

Reply:
@Alenar found another way to handle this ([here](https://github.com/input-output-hk/mithril/pull/1752/commits/3c819e659c43fba32a413da8637bf10e94f8e865#diff-2674a58fbdef4c48baa48bf5d619f381c0950f9540b2ab14f07a7ceb5a5d5762R21)) without adding an additional step specific to `macos-14`.

Reply:
Great!